### PR TITLE
UE pass on Partial views doc

### LINKED ISSUE #7322: Link to partial tag helper?
Would it be useful to link to the partial tag helper here?
https://docs.microsoft.com/en-us/aspnet/core/mvc/views/tag-helpers/built-in/partial-tag-helper?view=aspnetcore-2.1

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d670c64c-4030-312c-4355-ee518a23a56a
* Version Independent ID: c20d4db8-d403-9218-c39d-47e56c34e767
* Content: [Partial Views in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/mvc/views/partial?view=aspnetcore-2.1)
* Content Source: [aspnetcore/mvc/views/partial.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/mvc/views/partial.md)
* Product: **aspnet-core**
* GitHub Login: @ardalis
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by scottaddie**
Fixes #7322 

**Changes**
- Add Partial Tag Helper info (2.1+ only)
- Simplify & improve sample app and update to 2.1
- Add **Additional resources** section
- Apply Acrolinx suggestions
- Introduce headings in **Reference a partial view** section

[Internal Review Page](https://review.docs.microsoft.com/en-us/aspnet/core/mvc/views/partial?branch=pr-en-us-7417&view=aspnetcore-2.1)

### CODE REVIEW

File: aspnetcore/mvc/views/partial.md [hunk 1]
```diff
@@ -1,112 +1,152 @@
...
+ms.date: 07/02/2018
 uid: mvc/views/partial
 ---
-# Partial Views in ASP.NET Core
+# Partial views in ASP.NET Core
 
 By [Steve Smith](https://ardalis.com/), [Maher JENDOUBI](https://twitter.com/maherjend), [Rick Anderson](https://twitter.com/RickAndMSFT), and [Scott Sauber](https://twitter.com/scottsauber)
 
-ASP.NET Core MVC supports partial views, which are useful when you have reusable parts of web pages you want to share between different views.
+ASP.NET Core MVC supports partial views, which are useful for sharing reusable parts of web pages across different views.
```
Comment (Rick-Anderson) [inline aspnetcore/mvc/views/partial.md]: Consider adding --
For example, the default templates for a new web app with individual user accounts creates the following partials:

* _LoginPartial.cshtml:  Called in *_Layout.cshtml*.
* _CookieConsentPartial: Called in *_Layout.cshtml*.
*  *_ValidationScriptsPartial.cshtml*: ...

File: aspnetcore/mvc/views/partial.md [hunk 2]
```diff
@@ -1,112 +1,152 @@
...
-### Partial View Discovery
+[Partial](/dotnet/api/microsoft.aspnetcore.mvc.rendering.htmlhelperpartialextensions.partial) and [RenderPartial](/dotnet/api/microsoft.aspnetcore.mvc.rendering.htmlhelperpartialextensions.renderpartial) are the synchronous equivalents of `PartialAsync` and `RenderPartialAsync`, respectively. Use of the synchronous equivalents isn't recommended because there are scenarios in which they deadlock. Future releases won't contain the synchronous methods.
 
-When referencing a partial view, you can refer to its location in several ways:
+> [!IMPORTANT]
+> If your views need to execute code, use a [view component](xref:mvc/views/view-components) instead of a partial view.
+
+### Partial view discovery
+
+When referencing a partial view, you can refer to its location in several ways. For example:
```
Comment (Rick-Anderson) [inline aspnetcore/mvc/views/partial.md]: Feature the partial tag helper
    `<partial name="_CookieConsentPartial" />`

We need to show that and that's it's the preferred approach.

**Comment by scottaddie** [comment]
@Rick-Anderson Please take another look.

**Comment by Rick-Anderson** [comment]
@scottaddie  can you merge this then merge the live PR?

**Pull request merged**